components/post_view/embedded_forms/embedded_form/embedded_form.tsx

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+/* eslint-disable react/no-string-refs */
+
+import React, {CSSProperties} from 'react';
+
+import {AppBinding, PostEmbed} from 'mattermost-redux/types/apps';
+
+import {Post} from 'mattermost-redux/types/posts';
+
+import memoize from 'memoize-one';
+
+import * as Utils from 'utils/utils';
+import LinkOnlyRenderer from 'utils/markdown/link_only_renderer';
+import {TextFormattingOptions} from 'utils/text_formatting';
+
+import Markdown from 'components/markdown';
+import ShowMore from 'components/post_view/show_more';
+
+import ActionButton from '../action_button';
+import ActionMenu from '../action_menu';
+
+import {fillBindingsInformation} from 'utils/apps';
+
+type Props = {
+
+    /**
+     * The post id
+     */
+    post: Post;
+
+    /**
+     * The attachment to render
+     */
+    embed: PostEmbed;
+
+    /**
+     * Options specific to text formatting
+     */
+    options?: Partial<TextFormattingOptions>;
+
+    currentRelativeTeamUrl: string;
+}
+
+export default class EmbeddedForm extends React.PureComponent<Props> {
```

Comment:
One idea would be to use `EmbeddedFormComponent` on the client instead of `EmbeddedForm` on the client (or whatever PP struct name we come up with instead). and use `embeddedform(s)` instead of `embed(s)` as the supporting data model.